stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/EventHandlerImpl.kt

```diff
@@ -385,29 +406,42 @@ internal class EventHandlerImpl(
     }
 
     private suspend fun handleEventsInternal(events: List<ChatEvent>) {
-        events.sortedBy { it.createdAt }
-        updateOfflineStorageFromEvents(events)
+        val sortedEvents = events.sortedBy { it.createdAt }
+        updateOfflineStorageFromEvents(sortedEvents)
 
         // step 3 - forward the events to the active channels
+        sortedEvents
+            .filterIsInstance<CidEvent>()
+            .let { cidEvents ->
+                // handle CidEvents with non-null CIDs
```

Comment:
`CidEvent` was created to not need this kind of check. We shouldn't allow null value there